plugins/inputs/vsphere/endpoint.go

```diff
@@ -588,6 +591,17 @@ func (e *Endpoint) Collect(ctx context.Context, acc telegraf.Accumulator) error
 }
 
 func (e *Endpoint) chunker(ctx context.Context, f PushFunc, res *resourceKind, now time.Time, latest time.Time) {
+	maxMetrics := e.Parent.MaxQueryMetrics
+	if maxMetrics == 0 {
```

Comment:
Maybe check `if maxMetrics < 1`

Reply:
Look a couple of lines down at om.Query(...). The context is used there.

Reply:
Good point.

Reply:
Done